src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +35,113 @@
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        List<Prefix> getPrefixInArgs = getPrefixes(trimmedArgs);
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        if (getPrefixInArgs.isEmpty()) {
+            List<String> arguments = Arrays.asList(trimmedArgs.split(" "));
+            List<String> trimmedArguments = prepareArguments(arguments);
+            return new FindCommand(new AnyContainsKeywordsPredicate(trimmedArguments));
+        }
+
+        if (getPrefixInArgs.size() > 1) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        Prefix targetPrefix = getPrefixInArgs.get(0);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, targetPrefix);
+
+        if (!isAPrefixWithValue(argMultimap, targetPrefix)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        if (targetPrefix.equals(PREFIX_NAME)) {
+            List<String> name = argMultimap.getAllValues(PREFIX_NAME);
+            List<String> trimmedName = prepareArguments(name);
+            predicate = new NameContainsKeywordsPredicate(trimmedName);
+        } else if (targetPrefix.equals(PREFIX_PHONE)) {
+            List<String> phone = argMultimap.getAllValues(PREFIX_PHONE);
+            List<String> trimmedPhone = prepareArguments(phone);
+            predicate = new PhoneContainsKeywordsPredicate(trimmedPhone);
+        } else if (targetPrefix.equals(PREFIX_EMAIL)) {
+            List<String> email = argMultimap.getAllValues(PREFIX_EMAIL);
+            List<String> trimmedEmail = prepareArguments(email);
+            predicate = new EmailContainsKeywordsPredicate(trimmedEmail);
+        } else if (targetPrefix.equals(PREFIX_ADDRESS)) {
+            List<String> address = argMultimap.getAllValues(PREFIX_ADDRESS);
+            List<String> trimmedAddress = prepareArguments(address);
+            predicate = new AddressContainsKeywordsPredicate(trimmedAddress);
+        } else if (targetPrefix.equals(PREFIX_TAG)) {
+            List<String> tagList = argMultimap.getAllValues(PREFIX_TAG);
+            List<String> trimmedTagList = prepareArguments(tagList);
+            predicate = new TagContainsKeywordsPredicate(trimmedTagList);
+        } else {
+            throw new ParseException("Something bad has happened during parsing. Please contact the project team!");
+        }
+        return new FindCommand(predicate);
+    }
+
+    /**
+     * Returns a list of prefixes that can be found in the arguments list.
+     * @param args
+     */
+    private static List<Prefix> getPrefixes(String args) {
+        HashMap<String, Prefix> prefixMapping = new HashMap<>();
```

Comment:
Consider having this mapping be a static variable in `CliSyntax`.
(A static initialiser can be used to make a hashmap.)

Reply:
I like this idea